### ISSUE TITLE
installer build script

### ISSUE THREAD

**Issue by networkimprov** [networkimprov/arch-packages#50]
Near-term, I need a cross-compiling build shell script for u-boot. We can add this to our current u-boot branch "anvl-v2016.09" (High priority)

Long-term I plan to make a shell script which builds omap34usbboot + u-boot + rootfs, and creates a tarball so we can drop arch/* + images/{MLO,u-boot.bin} from installer repo. (Low priority)

**Comment by tmlind** [comment]
export ARCH
export CROSS_COMPILE=arm-linux-gnueabihf-
make omap3_anvl_config
make -j8

**Comment by networkimprov** [comment]
export ARCH=arm ?

**Comment by mranostay** [comment]
On Tue, Sep 20, 2016 at 10:00 PM, Liam <notifications@github.com> wrote:

> export ARCH=arm ?
>

Yeah that would be correct

**Comment by tmlind** [comment]
Heh yeah

**Comment by networkimprov** [comment]
@tmlind I need a build script for omap34usbboot, unless '$ make' is sufficient?

**Comment by networkimprov** [comment]
got omap34usbboot building with `TOOLCHAIN=arm-none-eabi- make`
It produces a lot of warnings, disconcerting :-(
This only builds x86_64 binaries. How to build i386?

**Comment by tmlind** [comment]
I've just built the i386 version by patching Makefile for HOST_CFLAGS := -g -O2 -Wall -m32. Yeah the usbboot part is not in good shape..

**Comment by networkimprov** [comment]
@tmlind, so no config other than TOOLCHAIN and HOST_CFLAGS?

**Comment by tmlind** [comment]
I don't think you even need to specify TOOLCHAIN.

**Comment by networkimprov** [comment]
I did as I have have arm-none-eabi-* which isn't the default.

**Comment by tmlind** [comment]
This you need to buld for the host, that's x86 typically.

**Comment by networkimprov** [comment]
Right, the makefile uses TOOLCHAIN for the omap binaries, not x86 binaries.

**Comment by networkimprov** [comment]
I can't build 32bit binaries out of the box on my x86_64 Arch Linux system (there's no 32bit glibc I can install quickly), so I'm inclined to drop i386 support in omap34usbboot/out/usbboot. Thoughts?

**Comment by tmlind** [comment]
Ah ok yeah I have multiarch support installed on my PC. At least I don't care about i386 support, most computers are x86_64 anyways.
